volatility3/framework/plugins/linux/proc.py

```diff
@@ -35,16 +40,138 @@ def get_requirements(cls):
                 element_type=int,
                 optional=True,
             ),
+            requirements.BooleanRequirement(
+                name="dump",
+                description="Extract listed memory segments",
+                default=False,
+                optional=True,
+            ),
+            requirements.ListRequirement(
+                name="address",
+                description="Process virtual memory address to include "
+                "(all other address ranges are excluded). This must be "
+                "a base address, not an address within the desired range.",
+                element_type=int,
+                optional=True,
+            ),
+            requirements.IntRequirement(
+                name="maxsize",
+                description="Maximum size for dumped VMA sections "
+                "(all the bigger sections will be ignored)",
+                default=cls.MAXSIZE_DEFAULT,
+                optional=True,
+            ),
         ]
 
+    @classmethod
+    def list_vmas(
+        cls,
+        task: interfaces.objects.ObjectInterface,
+        filter_func: Callable[
+            [interfaces.objects.ObjectInterface], bool
+        ] = lambda _: False,
+    ) -> Generator[interfaces.objects.ObjectInterface, None, None]:
+        """Lists the Virtual Memory Areas of a specific process.
+
+        Args:
+            task: task object from which to list the vma
+            filter_func: Function to take a vma and return True if it should be filtered out
+
+        Returns:
+            A list of vmas based on the task and filtered based on the filter function
+        """
+        if task.mm:
+            for vma in task.mm.get_mmap_iter():
+                if not filter_func(vma):
+                    yield vma
+
+    @classmethod
+    def vma_dump(
+        cls,
+        context: interfaces.context.ContextInterface,
+        task: interfaces.objects.ObjectInterface,
+        vma: interfaces.objects.ObjectInterface,
```

Comment:
I'd prefer this weren't passed as a complex type, but as simple integers.  I'm ok with one or two complex parameters where it makes sense (such as task), but this is literally just holding two numbers, and if someone wants to call this without having gone the route you expected, they'll need to construct complex objects just to pass in two numbers.  That should also shift where the AttributeError check occurs.

Reply:
I can't work out a good way to not pass vma as an object. I could pass the start address for a vma and then go through `task.mm.get_mmap_iter()` again to find it. or is there some way of accessing the right one without needing to go through the loop? I must be missing something sorry.

Reply:
Looks like you figured out what I was after.  5;)